### DISCUSSION TITLE
relative difference

### DISCUSSION THREAD

**Discussion by KOuadie** [mtex-toolbox/mtex#689] [Ask Anything]
Hello everyone,
I am a beginner level user. I have EBSD and XRD data, and my goal is to compare between the odfs/pole figures obtained with XRD and EBSD for the same material (dual-phase steel).
I have 6 EBSD datasets that I combined.
I would like to use a relative difference - means: (Value_EBSD - Value_XRD) / Value_EBSD so that I can get more or less a percentage deviation of XRD compared to EBSD. I think with this it should also be easier to compare all the results of the different combinations of EBSD measurements mentioned above. I'd also like to use the same scale for all plots which would be perfect for the comparison.
The problem is I can't divide the odfs, for some reason I always get an error message. Can anyone help me out with this?
I am using MatLAB R2020a and MTEX5.5.beta.3.
Thank you in advance for your help.
Best regards,
K. Ouadie

**Comment by kilir** [comment]
Hi,
for the moment, you need to evaluate your odfs on a grid (see `help SO3Grid`) and compute with those. The function `ODF/calcError` might also be interesting for you.

Cheers,
Rüdiger

**Reply by KOuadie** [reply]
Hello Rüdiger,
Thank you for your answer.
Unfortunately, I still wasn't able to achieve my goal. With my XRD data, I could easily evaluate my odfs on a grid using the SO3Grid function but it didn't work with the EBSD data. I used the scatter function with the EBSD data but even then I couldn't find the relative difference between the EBSD and the XRD.
As for the ODF/calcError function, it is good but it's not exactly what I'm looking for.
Do you have any other suggestions?
Sincerely,
K. Ouadie

**Reply by kilir** [reply]
Hi,
I'm not entirely sure what you mean by "didn't work on the EBSD data". You calculate an odf from XRD and you evaluate the odf values on a grid but when you calculate an odf from the EBSd data, you cannot evaluate the odf values?
You may have to tell us a little bit about what you did, preferntially the code you used.
Cheers,
Rüdiger

**Reply by KOuadie** [reply]
Hi,
here are my codes : 
`%% Import Script for PoleFigure Data
%
% This script was automatically created by the import wizard. You should
% run the whoole script or parts of it in order to import your data. There
% is no problem in making any changes to this script.

%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = crystalSymmetry('432', [2.9 2.9 2.9], 'mineral', 'Ferrite', 'color', [0.53 0.81 0.98]);

% specimen symmetry
SS = specimenSymmetry('1');

% plotting convention
setMTEXpref('xAxisDirection','north');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% path to files
pname = 'C:\Users\Khawla\Desktop\Stage LABO\XRD DATA';

% which files to be imported
fname = {...
  [pname '\20200620_Selina-Neuhaus_DH_Texture - S {110} - krz_Blechviertel_SebastianSlawik.xrdml'],...
  [pname '\20200620_Selina-Neuhaus_DH_Texture - S {200} - krz_Blechviertel_SebastianSlawik.xrdml'],...
  [pname '\20200620_Selina-Neuhaus_DH_Texture - S {211} - krz_Blechviertel_SebastianSlawik.xrdml'],...
  [pname '\20200620_Selina-Neuhaus_DH_Texture - S {220} - krz_Blechviertel_SebastianSlawik.xrdml'],...
  };

% background
pname = 'C:\Users\Khawla\Desktop\Stage LABO\XRD DATA\background measurements\attachments';
fname_bg = {...
  [pname '\20200620_Selina-Neuhaus_DH_Texture - B {110} - krz_Blechviertel_SebastianSlawik.xrdml'],...
  [pname '\20200620_Selina-Neuhaus_DH_Texture - B {200} - krz_Blechviertel_SebastianSlawik.xrdml'],...
  [pname '\20200620_Selina-Neuhaus_DH_Texture - B {211} - krz_Blechviertel_SebastianSlawik.xrdml'],...
  [pname '\20200620_Selina-Neuhaus_DH_Texture - B {220} - krz_Blechviertel_SebastianSlawik.xrdml'],...
  };

%% Specify Miller Indice

h = { ...
  Miller(1,1,0,CS),...
  Miller(2,0,0,CS),...
  Miller(2,1,1,CS),...
  Miller(2,2,0,CS),...
  };

%% Import the Data

% create a Pole Figure variable containing the data
pf = PoleFigure.load(fname,h,CS,SS,'interface','xrdml');

% background
pf_bg = PoleFigure.load(fname_bg,h,CS,SS,'interface','xrdml');

% correct data
pf = correct(pf,'bg',pf_bg);
plot(pf)
odf_FER_XRD = calcODF(pf)
odf_FER_XRD = rotate(odf_FER_XRD,rotation.byAxisAngle(xvector,-90*degree));
odf_FER_XRD = rotate(odf_FER_XRD,rotation.byAxisAngle(yvector,0*degree));
odf_FER_XRD = rotate(odf_FER_XRD,rotation.byAxisAngle(zvector,-90*degree));

% evaluating odfs on grid
S3G = regularSO3Grid(CS)

`

`%% Import Script for EBSD Data
%
% This script was automatically created by the import wizard. You should
% run the whoole script or parts of it in order to import your data. There
% is no problem in making any changes to this script.

%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('432', [2.9 2.9 2.9], 'mineral', 'Ferrite', 'color', [0.53 0.81 0.98]),...
  crystalSymmetry('432', [3.6 3.6 3.6], 'mineral', 'Austenite', 'color', [0.56 0.74 0.56])};

% plotting convention
setMTEXpref('xAxisDirection','north');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% path to files
pname = 'C:\Users\Khawla\Desktop\Stage LABO\EBSD DATA';

% which files to be imported
fname = [pname '\Quer_A_S0002927b_Mod.ang'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ang',...
  'convertEuler2SpatialReferenceFrame');
ebsdA_b = ebsd;
%% EBSD file A_a
%% Import Script for EBSD Data
%
% This script was automatically created by the import wizard. You should
% run the whoole script or parts of it in order to import your data. There
% is no problem in making any changes to this script.

%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('432', [2.9 2.9 2.9], 'mineral', 'Ferrite', 'color', [0.53 0.81 0.98]),...
  crystalSymmetry('432', [3.6 3.6 3.6], 'mineral', 'Austenite', 'color', [0.56 0.74 0.56])};

% plotting convention
setMTEXpref('xAxisDirection','north');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% path to files
pname = 'C:\Users\Khawla\Desktop\Stage LABO\EBSD DATA';

% which files to be imported
fname = [pname '\Quer_A_S0002927a_Mod.ang'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ang',...
  'convertEuler2SpatialReferenceFrame');
ebsdA_a = ebsd;
%% EBSD B_b
%% Import Script for EBSD Data
%
% This script was automatically created by the import wizard. You should
% run the whoole script or parts of it in order to import your data. There
% is no problem in making any changes to this script.

%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('432', [2.9 2.9 2.9], 'mineral', 'Ferrite', 'color', [0.53 0.81 0.98]),...
  crystalSymmetry('432', [3.6 3.6 3.6], 'mineral', 'Austenite', 'color', [0.56 0.74 0.56])};

% plotting convention
setMTEXpref('xAxisDirection','north');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% path to files
pname = 'C:\Users\Khawla\Desktop\Stage LABO\EBSD DATA';

% which files to be imported
fname = [pname '\Laengs_B_S0002937b_Mod.ang'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ang',...
  'convertEuler2SpatialReferenceFrame');
ebsdB_b = ebsd;
%% EBSD B_a
%% Import Script for EBSD Data
%
% This script was automatically created by the import wizard. You should
% run the whoole script or parts of it in order to import your data. There
% is no problem in making any changes to this script.

%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('432', [2.9 2.9 2.9], 'mineral', 'Ferrite', 'color', [0.53 0.81 0.98]),...
  crystalSymmetry('432', [3.6 3.6 3.6], 'mineral', 'Austenite', 'color', [0.56 0.74 0.56])};

% plotting convention
setMTEXpref('xAxisDirection','north');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% path to files
pname = 'C:\Users\Khawla\Desktop\Stage LABO\EBSD DATA';

% which files to be imported
fname = [pname '\Laengs_B_S0002937a_Mod.ang'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ang',...
  'convertEuler2SpatialReferenceFrame');
ebsdB_a = ebsd;
%% EBSD C_b
%% Import Script for EBSD Data
%
% This script was automatically created by the import wizard. You should
% run the whoole script or parts of it in order to import your data. There
% is no problem in making any changes to this script.

%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('432', [2.9 2.9 2.9], 'mineral', 'Ferrite', 'color', [0.53 0.81 0.98]),...
  crystalSymmetry('432', [3.6 3.6 3.6], 'mineral', 'Austenite', 'color', [0.56 0.74 0.56])};

% plotting convention
setMTEXpref('xAxisDirection','north');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% path to files
pname = 'C:\Users\Khawla\Desktop\Stage LABO\EBSD DATA';

% which files to be imported
fname = [pname '\Flach_C_S0002956b_Mod.ang'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ang',...
  'convertEuler2SpatialReferenceFrame');
ebsdC_b = ebsd;
%% EBSD C_a
%% Import Script for EBSD Data
%
% This script was automatically created by the import wizard. You should
% run the whoole script or parts of it in order to import your data. There
% is no problem in making any changes to this script.

%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('432', [2.9 2.9 2.9], 'mineral', 'Ferrite', 'color', [0.53 0.81 0.98]),...
  crystalSymmetry('432', [3.6 3.6 3.6], 'mineral', 'Austenite', 'color', [0.56 0.74 0.56])};

% plotting convention
setMTEXpref('xAxisDirection','north');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% path to files
pname = 'C:\Users\Khawla\Desktop\Stage LABO\EBSD DATA';

% which files to be imported
fname = [pname '\Flach_C_S0002956a_Mod.ang'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ang',...
  'convertEuler2SpatialReferenceFrame');

ebsdC_a = ebsd;

%% Combined data

ebsd=[ebsdA_b ebsdA_a ebsdB_b ebsdB_a ebsdC_b ebsdC_a];

odf_FER_ebsd = calcDensity(ebsd('Ferrite').orientations)

% evaluating odfs on grid

S3G = regularSO3Grid(CS)

`

Thank you in advance for your help.
K. Ouadie

**Reply by kilir** [reply]
Ok, you were missing the actual step to evaluate the ODF at the grid points.
```
odf_values_ebsd =  eval(odf_FER_ebsd,S3G)
odf_values_xrd = eval(odf_FER_XRD,S3G)
```
Cheers,
Rüdiger

**Reply by KOuadie** [reply]
Hi,
Thank you so much Rüdiger, it worked!
Sincerely,
K.Ouadie
